frontend/test/metabase/scenarios/native/snippets/snippet-permissions.cy.spec.js

```diff
@@ -158,20 +159,74 @@ describeWithToken("scenarios > question > snippets", () => {
     cy.findByText("snippet 1");
   });
 
-  it("should not display snippet folder as part of collections (metabase#14907)", () => {
-    cy.server();
-    cy.route("GET", "/api/collection/root").as("collections");
-
-    cy.request("POST", "/api/collection", {
-      name: "Snippet Folder",
-      description: null,
-      color: "#509EE3",
-      parent_id: null,
-      namespace: "snippets",
+  describe("existing snippet folder", () => {
+    beforeEach(() => {
+      cy.intercept("GET", "/api/collection/root").as("collections");
+
+      cy.request("POST", "/api/collection", {
+        name: "Snippet Folder",
+        description: null,
+        color: "#509EE3",
+        parent_id: null,
+        namespace: "snippets",
+      });
+    });
+
+    it("should not display snippet folder as part of collections (metabase#14907)", () => {
+      cy.visit("/collection/root");
+
+      cy.wait("@collections");
+      cy.findByText("Snippet Folder").should("not.exist");
     });
 
-    cy.visit("/collection/root");
-    cy.wait("@collections");
-    cy.findByText("Snippet Folder").should("not.exist");
+    it.skip("shouldn't update root permissions when changing permissions on a created folder (metabase#17268)", () => {
+      cy.intercept("PUT", "/api/collection/graph").as("updatePermissions");
+
+      openNativeEditor();
+      cy.icon("snippet").click();
+
+      cy.findByTestId("sidebar-right").within(() => {
+        cy.findByText("Snippet Folder")
+          .next()
+          .find(".Icon-ellipsis")
+          .click({ force: true });
+      });
+
+      cy.findByText("Change permissions").click();
+
+      // Update permissions for "All users"
+      modal().within(() => {
+        cy.findAllByRole("grid")
+          .last()
+          .find(".Icon-close")
+          .first()
+          .click();
+      });
```

Comment:
@alxnddr when you work on fixing this issue, can you please think of a better way to "mark" these permission icons. If nothing comes to mind, at least let's add some test-ids there.

Reply:
Actually, I think it is fine for now keeping in mind that we will change the permissions UI soon which will lead to changing this test as well